src/components/Button/Button.tsx

```diff
@@ -206,14 +212,20 @@ function ButtonComponent(
   ref?: React.Ref<HTMLButtonElement & HTMLAnchorElement>
 ): ReturnType;
 function ButtonComponent(
-  { children, icon: Icon, ...props }: ButtonProps,
+  { children, icon: Icon, tracking, ...props }: ButtonProps,
   ref?: React.Ref<HTMLButtonElement & HTMLAnchorElement>
 ): ReturnType {
   const { Link } = useComponents();
   const LinkButton = BaseButton.withComponent(Link);
   const ButtonElement = props.href ? LinkButton : BaseButton;
+  const handleClick = useClickTracker<MouseEvent<any>>(
```

Comment:
Can't we be more specific with the `MouseEvent` generics?, We know it's either a LinkButton or BaseButton. Does it make sense?

Reply:
We tried, but if we make it more specific, TypeScript complains. It's tricky because of the function overloads that we use in the Button component.